src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,78 @@
+package parser;
+
+import command.Command;
+import command.DeadlineCommand;
+import command.DeleteCommand;
+import command.DoneCommand;
+import command.EventCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.ToDoCommand;
+import exception.DukeException;
+
+import java.util.Scanner;
+
+/**
+ * Deals with making sense of user command.
+ */
+public class Parser {
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param userInput String of input of user
+     * @return the command based on user input
+     * @throws DukeException if user input is invalid
+     */
+    public Command parse(String userInput) throws DukeException {
+        String[] separatedInputs = userInput.trim().split("\\s+");
+        switch (separatedInputs[0].toLowerCase()) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "done":
+            //fallthrough
+        case "delete":
+            if (separatedInputs.length <= 1) {
```

Comment:
I like that input validation is performed as early as possible so that `Command` implementations can be simplified. 👍